### PR TITLE
FSharp: Add chained expressions guidance.

### LINKED ISSUE #688: [style-guide] Chain of (fluent) calls
We faced an issue that current version of Fantomas format one liner
```fsharp
Cache.providedTypes.GetOrAdd(cacheKey, addCache).Value
```
like this
```fsharp
Cache
    .providedTypes
    .GetOrAdd(
        cacheKey,
        addCache
    )
    .Value
```

In my opinion such formation is very verbose and @nojaf [mentioned](https://github.com/fsprojects/SwaggerProvider/pull/195#discussion_r913656270) that there is no guidance in style guide how to format such code.

In my opinion formatting should be similar to [pipeline expressions](https://github.com/dotnet/docs/blob/main/docs/fsharp/style-guide/formatting.md#formatting-pipeline-expressions) and look at least like this

```fsharp
Cache
    .providedTypes
    .GetOrAdd(cacheKey, addCache)
    .Value
```

### PR DESCRIPTION
**Pull request by nojaf**
## Summary

Add guidance for chained expressions to the F# style guide.

Fixes https://github.com/fsharp/fslang-design/issues/688

//cc @dsyme

### CODE REVIEW

File: docs/fsharp/style-guide/formatting.md [hunk 1]
```diff
@@ -1216,6 +1216,102 @@ In single-line expressions the delimiter symbols should be placed on the same li
...
+```
+
+The leading link can be composed out of multiple links if they are simple identifiers.
+For example, the addition of a fully qualified namespace.
+
+```fsharp
+// ✔️ OK
+Microsoft.Extensions.Hosting.Host
+    .CreateDefaultBuilder(args)
+    .ConfigureWebHostDefaults(fun webBuilder -> webBuilder.UseStartup<Startup>())
```
Comment (auduchinok) [inline docs/fsharp/style-guide/formatting.md:1246]: Why can't it contain method calls too?
The following code should be allowed by de style guide:

```fsharp
Microsoft.Extensions.GetHosting().Host
     .CreateDefaultBuilder(args)
     .ConfigureWebHostDefaults(fun webBuilder -> webBuilder.UseStartup<Startup>())
```

File: docs/fsharp/style-guide/formatting.md [hunk 2]
```diff
@@ -1216,6 +1216,102 @@ In single-line expressions the delimiter symbols should be placed on the same li
...
+        ())
+
+// ❌ Not OK, formatting tools will reformat to the above
+builder
+    .WithEnvironment()
+    .WithLogger(
+        fun loggerConfiguration ->
+        // ...
+        ()
+    )
```
Comment (auduchinok) [inline docs/fsharp/style-guide/formatting.md:1312]: Let's format the closing paren the same way as in the example above, so it's clear that paren placement is not related to lambda parameters?

File: docs/fsharp/style-guide/formatting.md (hunk 1 above)
Comment (nojaf) [reply docs/fsharp/style-guide/formatting.md:1246]: @auduchinok could you please join the conversation over at https://github.com/fsharp/fslang-design/issues/688 when it is taking place? This is frustrating that you question the decision of something that was labelled `style-guide-decided`. In the future please, subscribe to that repository and engage when the solution is being discussed.

To answer your question, the reasoning was that having the method calls always at the end would be more readable and make it clear when invocations are happening.

File: docs/fsharp/style-guide/formatting.md (hunk 2 above)
Comment (nojaf) [reply docs/fsharp/style-guide/formatting.md:1312]: That would be inconsistent with the existing lambda rules.
If the lambda was outside of a chain the style guide now says it should format like:
```fsharp
WithLogger(fun loggerConfiguration ->
    // ...
    ())
```

See https://learn.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-lambda-expressions

Comment (auduchinok) [reply docs/fsharp/style-guide/formatting.md:1312]: I mean changing the wrong sample, so it's the same with the correct one in this regard.

Comment (nojaf) [reply docs/fsharp/style-guide/formatting.md]: Apologies, I misunderstood, good points. Thanks!

File: docs/fsharp/style-guide/formatting.md (hunk 1 above)
Comment (auduchinok) [reply docs/fsharp/style-guide/formatting.md:1246]: >  In the future please, subscribe to that repository and engage when the solution is being discussed.

I am subscribed to it.

> @auduchinok could you please join the conversation over at https://github.com/fsharp/fslang-design/issues/688 when it is taking place?

Yes I could. I've seen it, and this example was not discussed there. I'm commenting a specific place in the PR in the appropriate place.

**Review by dsyme — APPROVED** [review]
